sklearn/utils/extmath.py

```diff
@@ -1030,8 +1030,10 @@ def _safe_accumulator_op(op, x, *args, **kwargs):
     result
         The output of the accumulator function passed to this function.
     """
-    if np.issubdtype(x.dtype, np.floating) and x.dtype.itemsize < 8:
-        result = op(x, *args, **kwargs, dtype=np.float64)
+    from ..utils._array_api import isdtype, get_namespace
+    xp, _ = get_namespace(x)
+    if isdtype(x.dtype, "real floating", xp=xp) and x.dtype in (xp.float32, xp.float64):  # what about int, etc.?
```

Comment:
Here is actually an error, I guess the second condition should be `x.dtype in (xp.float16, xp.float32)`.

Reply:
Since `x.dtype.itemsize < 8` is not an Array API compatible way to assess the precision level of a dtype, we could instead do `xp.finfo(x.dtype).bits < 64`.

Reply:
this should be solved